crypto/evp/evp_lib.c

```diff
@@ -44,13 +44,15 @@ int EVP_CIPHER_asn1_to_param(EVP_CIPHER_CTX *c, ASN1_TYPE *type)
 int EVP_CIPHER_get_asn1_iv(EVP_CIPHER_CTX *ctx, ASN1_TYPE *type)
 {
     int i = 0;
-    unsigned int l;
+    int l;
 
     if (type != NULL) {
         unsigned char iv[EVP_MAX_IV_LENGTH];
 
         l = EVP_CIPHER_CTX_get_iv_length(ctx);
-        if (!ossl_assert(l <= sizeof(iv)))
+        if (l < 0)
+            return -1;
+        if (!ossl_assert((long unsigned int)l <= sizeof(iv)))
```

Comment:
That is a strange cast ordering - and given that "iv" is declared right above you could avoid the sizeof.
I'd tend to have cast the sizeof to int rather than pointlessly promote the int ... but that is a style issue.

Reply:
I was focused on making sure l was typed as a signed int, since the return value from EVP_CIPHER_CTX_get_iv_length can be negtive if cipher isn't set yet (or for several other reasons).  Honestly, I think there were several issues with that function if EVP_CIPHER_CTX_get_iv_length returns -1, as the automatic conversion would turn a negative value into a huge positive value.